src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -623,8 +650,158 @@ private static MultiTermQuery.RewriteMethod getRewriteMethod(
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) prefixQuery);
+      case REGEXPQUERY:
+        RegexpQuery regexpQuery = getRegexpQuery(protoSpanMultiTermQuery);
+        return new SpanMultiTermQueryWrapper<>(regexpQuery);
+      case TERMRANGEQUERY:
+        TermRangeQuery termRangeQuery =
+            getTermRangeQuery(protoSpanMultiTermQuery.getTermRangeQuery());
+        return new SpanMultiTermQueryWrapper<>(termRangeQuery);
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported Span Multi Query Term Wrapper: " + protoSpanMultiTermQuery);
+    }
+  }
+
+  private static FuzzyQuery getFuzzyQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery) {
+    com.yelp.nrtsearch.server.grpc.FuzzyQuery protoFuzzyQuery =
+        protoSpanMultiTermQuery.getFuzzyQuery();
+    Term term = new Term(protoFuzzyQuery.getField(), protoFuzzyQuery.getText());
+
+    int maxEdits =
+        protoFuzzyQuery.hasMaxEdits() ? protoFuzzyQuery.getMaxEdits() : FuzzyQuery.defaultMaxEdits;
+
+    int prefixLength =
+        protoFuzzyQuery.hasPrefixLength()
+            ? protoFuzzyQuery.getPrefixLength()
+            : FuzzyQuery.defaultPrefixLength;
+
+    int maxExpansions =
+        protoFuzzyQuery.hasMaxExpansions()
+            ? protoFuzzyQuery.getMaxExpansions()
+            : FuzzyQuery.defaultMaxExpansions;
+
+    // Set the default transpositions to true, if it is not provided.
+    boolean transpositions =
+        protoFuzzyQuery.hasTranspositions()
+            ? protoFuzzyQuery.getTranspositions()
+            : FuzzyQuery.defaultTranspositions;
+
+    FuzzyQuery fuzzyQuery =
+        new FuzzyQuery(term, maxEdits, prefixLength, maxExpansions, transpositions);
+    fuzzyQuery.setRewriteMethod(
+        getRewriteMethod(protoFuzzyQuery.getRewrite(), protoFuzzyQuery.getRewriteTopTermsSize()));
+    return fuzzyQuery;
+  }
+
+  private static RegexpQuery getRegexpQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery) {
+
+    com.yelp.nrtsearch.server.grpc.RegexpQuery protoRegexpQuery =
+        protoSpanMultiTermQuery.getRegexpQuery();
+
+    Term term = new Term(protoRegexpQuery.getField(), protoRegexpQuery.getText());
+
+    int flags =
+        switch (protoRegexpQuery.getFlag()) {
+          case REGEXP_ALL -> RegExp.ALL;
+          case REGEXP_ANYSTRING -> RegExp.ANYSTRING;
+          case REGEXP_AUTOMATON -> RegExp.AUTOMATON;
+          case REGEXP_COMPLEMENT -> RegExp.COMPLEMENT;
+          case REGEXP_EMPTY -> RegExp.EMPTY;
+          case REGEXP_INTERSECTION -> RegExp.INTERSECTION;
+          case REGEXP_INTERVAL -> RegExp.INTERVAL;
+          case REGEXP_NONE -> RegExp.NONE;
+          default -> RegExp.ALL;
+        };
+
+    int maxDeterminizedStates =
+        protoRegexpQuery.hasMaxDeterminizedStates()
+            ? protoRegexpQuery.getMaxDeterminizedStates()
+            : Operations.DEFAULT_DETERMINIZE_WORK_LIMIT;
```

Comment:
The nomenclature has been changed from max number of states to work limit. The change is there in the constructor value as well. This constant was `DEFAULT_MAX_DETERMINIZED_STATES` in master but had to be changed to `DEFAULT_DETERMINIZE_WORK_LIMIT`. The value of the constant is still 10000.
I'm not changing anything in the protobuf value yet, but if this query is not being used it might be a good idea to do so. cc: @vim345